src/headingbuttonsui.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module heading/headingbuttonsui
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+
+import { getLocalizedOptions } from './utils';
+import iconHeading1 from '../theme/icons/heading1.svg';
+import iconHeading2 from '../theme/icons/heading2.svg';
+import iconHeading3 from '../theme/icons/heading3.svg';
+
+const defaultIcons = {
+	heading1: iconHeading1,
+	heading2: iconHeading2,
+	heading3: iconHeading3
+};
+
+/**
+ * HeadingButtonsUI class creates a set of UI buttons that can be used instead of drop down component.
+ * It is not enabled by default when using {@link module:heading/heading~Heading heading plugin}, and needs to be
+ * added manually to the editor configuration.
+ *
+ * Plugin introduces button UI elements, which names are same as `model` property from {@link module:heading/heading~HeadingOption}.
+ *
+ *		ClassicEditor
+ *			.create( {
+ *				plugins: [ ..., Heading, Paragraph, HeadingButtonsUI, ParagraphButtonUI ]
+ *				heading: {
+ *					options: [
+ *						{ model: 'paragraph', title: 'Paragraph', class: 'ck-heading_paragraph' },
+ *						{ model: 'heading1', view: 'h2', title: 'Heading 1', class: 'ck-heading_heading1' },
+ *						{ model: 'heading2', view: 'h3', title: 'Heading 2', class: 'ck-heading_heading2' },
+ *						{ model: 'heading3', view: 'h4', title: 'Heading 3', class: 'ck-heading_heading3' }
+ *					]
+ * 				},
+ * 				toolbar: [ 'paragraph', 'heading1', 'heading2', 'heading3' ]
+ *			} )
+ *			.then( ... )
+ *			.catch( ... );
+ *
+ * NOTE: Paragraph button is defined in {@link module:paragraph/paragraphbuttonui~ParagraphButtonUI} plugin that needs
+ * to be loaded manually as well.
+ *
+ * It is possible to use custom icons by providing `icon` config option provided in {@link module:heading/heading~HeadingOption}.
+ * For the default configuration standard icons are used.
```

Comment:
You should add a note here that icons are defined only for heading 1 - 3 and custom icon is needed for other headers. It might be not clear what "default configuration" means.